client/src/components/requests/RequestedList.tsx

```diff
@@ -0,0 +1,169 @@
+import React from "react";
+import {
+    Card,
+    Container,
+    Header, Icon, Item,
+    Label, Loader, Message,
+    Step
+} from "semantic-ui-react";
+import {RequestedItem} from "../../types/Hardware";
+import {useQuery} from "@apollo/react-hooks";
+import {DESK_REQUESTS} from "../util/graphql/Queries";
+import {Request} from "../../types/Request";
+import ItemAndQuantity from "../desk/ItemAndQuantity";
+
+interface RequestedListProps {
+    requestedItemsList: RequestedItem[]
+}
+
+function RequestedList({requestedItemsList}: RequestedListProps) {
+    const {loading, error, data} = useQuery(DESK_REQUESTS, {
+        pollInterval: 500
+    });
+    if (loading) {
+        return (
+            <Loader active inline="centered" content="Please wait..."/>
+        );
+    }
+
+    if (error) {
+        return (
+            <Message
+                title="View-only inventory"
+                warning icon>
+                <Icon name="warning sign"/>
+                {error.message}
+            </Message>);
+    }
+
+    let steps = (
+        <Step.Group>
+            <Step>
+                <Step.Content>
+                    <Step.Title>Loading steps</Step.Title>
+                </Step.Content>
+            </Step>
+        </Step.Group>
+    );
+
+    if (data.requests.length === 0) {
+        return (
+            <Container textAlign="center">
+                <Header>
+                    You haven't requested any hardware yet. To request an item,
+                    select the quantity and click on the blue Request button.
+                </Header>
+            </Container>
+        );
+    }
+
+    if (data.requests.length > 0) {
+        return data.requests.map((r: Request, index: number) => {
+            if (r.status === 'SUBMITTED') {
```

Comment:
A lot of the code here is repetitive.  Could you have one if statement for Submitted, Approved, or Ready for Pickup, and then render one set of steps for those and conditionally indicate which one is `active`?  Same thing for the other groups

Ex:

```jsx
if (a || b || c) {
    return <blah> <Element active={a === 's'} /> </blah>;
}
```